lib/jsdom/living/geometry/DOMRectReadOnly-impl.js

```diff
@@ -0,0 +1,95 @@
+"use strict";
+
+class DOMRectReadOnlyImpl {
+  constructor(globalObject, [x = 0, y = 0, width = 0, height = 0], privateData) {
+    this._globalObject = globalObject;
+    this._x = x;
+    this._y = y;
+    this._width = width;
+    this._height = height;
+
+    // Reflection mode for SVG's viewBox.
+    this._reflectedElement = privateData.reflectedElement;
+    this._reflectedAttribute = privateData.reflectedAttribute;
+    this._parser = privateData.parser;
+  }
+
+  get x() {
+    if (this._reflectedElement) {
+      const attr = this._reflectedElement.getAttributeNS(null, this._reflectedAttribute);
+      return this._parser(attr).x;
+    }
+    return this._x;
+  }
+
+  get y() {
+    if (this._reflectedElement) {
+      const attr = this._reflectedElement.getAttributeNS(null, this._reflectedAttribute);
+      return this._parser(attr).y;
+    }
+    return this._y;
+  }
+
+  get width() {
+    if (this._reflectedElement) {
+      const attr = this._reflectedElement.getAttributeNS(null, this._reflectedAttribute);
+      return this._parser(attr).width;
+    }
+    return this._width;
+  }
+
+  get height() {
+    if (this._reflectedElement) {
+      const attr = this._reflectedElement.getAttributeNS(null, this._reflectedAttribute);
+      return this._parser(attr).height;
+    }
+    return this._height;
+  }
+
+  get top() {
+    const { height, y } = this;
+    if (isNaN(height)) {
```

Comment:
This NaN check is not in the spec, if I am reading correctly. Another spec bug to file? Or maybe the same one, if it can only happen for the reflection case?

Reply:
No this is different from the reflection. In fact this is much worse. Essentially, the question is `new DOMRectReadOnly(0, 0, NaN, NaN).top/bottom/left/right` NaN, or 0?

The spec says one of them should be the result of min(0, 0 + NaN), but doesn't say how to compare NaN with actual numbers. Implementations don't have particular handling either, relying instead of platform primitives like `std::min`. The result, is that Chrome and Firefox on Linux (as well as Chrome on macOS) seem to prefer non-NaNs, but Safari/WebKit on Linux and macOS prefer NaNs. But theoretically the behavior could vary between browsers but also between different operating systems, compiler version, standard library, compiler flags in use, the function (min vs max), etc.

A spec bug seems definitely appropriate, but I'm not sure about what we should do here. Too many choices! ;)

Reply:
Well, guess there's already one from two years ago: https://github.com/w3c/fxtf-drafts/issues/222

Reply:
Interesting. I'd say we should just let the NaNs propagate according to the Math.min/max behavior. Some of the later comments there are leaning in that direction too. And it seems like you've gone in that direction as well, per https://github.com/w3c/fxtf-drafts/pull/395 :).